zarr/attrs.py

```diff
@@ -10,23 +10,42 @@
 
 class Attributes(MutableMapping):
 
-    def __init__(self, store, key='.zattrs', read_only=False,
+    def __init__(self, store, key='.zattrs', read_only=False, cache=True,
                  synchronizer=None):
         self.store = store
         self.key = key
         self.read_only = read_only
+        self.cache = cache
+        self._cached_asdict = None
         self.synchronizer = synchronizer
 
+    def _get(self):
+        if self.key in self.store:
+            d = json.loads(text_type(self.store[self.key], 'ascii'))
```

Comment:
As a general note, in the future for remote datasets we might want to minimize multiple touches of `self.store`.  I can imagine wanting to change things like this to code like the following instead:

```python
try:
    data = self.store[self.key[
except KeyError:
    d = {}
else:
    d = json.loads(text_type(data))
```

This is just a general comment though, not something that should necessarily be done here.  I'll bring it up again if we find this a challenge when profiling.

Reply:
Good suggestion, asking for permission is a hard habit to break :-)